Change wording of hostname providers in the status

remicalixte opened the pull request:
### What does this PR do?

Change wording of hostname providers in the status

### Motivation

We’re showing as `error`s things that are completely expected (for example: `error: configuration/environment: host name is empty` or `error: gce: unable to retrieve hostname from GCE: Get http://169.254.169.254/computeMetadata/v1/instance/hostname: dial tcp 169.254.169.254:80: connect: no route to host`) which can be misleading. Show them as `unused hostname providers` instead

remicalixte committed:
Change wording of hostname providers in the status

derekwbrown commented on Dockerfiles/cluster-agent/dist/templates/header.tmpl:
```diff
@@ -28,13 +28,17 @@
-    error: {{$value}}
-    {{- end}}
+      {{- if ne $name "all" }}
+      {{$name}}: {{$value}}
```
nit - indent

olivielpeau replied:
this weird indentation is necessary for the resulting status page to have the correct indentation (problem of code indentation vs. indentation of what's actually being displayed)

derekwbrown commented on pkg/status/dist/templates/header.tmpl:
```diff
@@ -44,11 +44,15 @@ NOTE: Changes made to this template should be reflected on the following templat
-    error: {{$value}}
-    {{- end}}
+      {{- if ne $name "all" }}
+      {{$name}}: {{$value}}
```
same

olivielpeau replied:
(same response as above)

derekwbrown reviewed (dismissed):
if you make the formatting changes, go ahead and check in as is.

olivielpeau reviewed (commented):
Overall LGTM, I've tested the Agent's status page and GUI status page and both look good. Going to push a couple of additional cosmetic fixes

olivielpeau committed:
[hostname] Fix error message from AWS provider and typo

The error message wasn't reflecting what our EC2 hostname provider was
actually doing, fix that + fix small typo `host name` -> `hostname`

truthbk reviewed (approved):
I see nothing wrong with this 👍

arbll commented on Dockerfiles/cluster-agent/dist/templates/header.tmpl:
```diff
@@ -28,13 +28,17 @@
-      hostname provider: {{.hostnameStats.provider}}
+    hostname provider: {{.hostnameStats.provider}}
+  {{- if gt (len .hostnameStats.errors) 0 }}
+    unused hostname providers:
```
Maybe unavailable instead of unused

olivielpeau replied:
sometimes the agent logic chooses not to use a hostname provider even it it's available, so I have a preference for unused

arbll reviewed (approved):
LGTM, just a proposal for a different choice of word, feel free to ignore
